### PR TITLE
remove unsupported save,load,read,write from api docs for knn estimat…

### PR DESCRIPTION
**Pull request by eordentlich**
…or, model classes

### CODE REVIEW

**Comment by eordentlich** [comment]
build

File: python/src/spark_rapids_ml/knn.py [hunk 1]
```diff
@@ -338,16 +338,31 @@ def _get_cuml_fit_func(self, dataset: DataFrame) -> Callable[  # type: ignore
...
     def read(cls) -> MLReader:
+        """Unsupported."""
         raise NotImplementedError(
             "NearestNeighbors does not support saving/loading, just re-create the estimator."
         )
 
+    def save(self, path: str) -> None:
+        """Unsupported."""
+        raise NotImplementedError(
+            "ApproximateNearestNeighbors does not support saving/loading, just re-create the estimator."
```
Comment (lijinf2) [inline python/src/spark_rapids_ml/knn.py:356]: Seems need to replace "ApproximateNearestNeighbors" with "NearestNeighbors", as the function belongs to Nearestneighbors?

File: python/src/spark_rapids_ml/knn.py [hunk 2]
```diff
@@ -338,16 +338,31 @@ def _get_cuml_fit_func(self, dataset: DataFrame) -> Callable[  # type: ignore
...
+        )
+
+    @classmethod
+    def load(cls, path: str) -> MLReader:
+        """Unsupported."""
+        raise NotImplementedError(
+            "ApproximateNearestNeighbors does not support saving/loading, just re-create the estimator."
```
Comment (lijinf2) [inline python/src/spark_rapids_ml/knn.py:363]: Same here. Seems the load function belongs to the NearestNeighbor class.

File: python/src/spark_rapids_ml/knn.py [hunk 3]
```diff
@@ -442,16 +457,31 @@ def exactNearestNeighborsJoin(
...
+        """Unsupported."""
         raise NotImplementedError(
             f"{self.__class__} does not support saving/loading, just re-fit the estimator to re-create a model."
         )
 
     @classmethod
     def read(cls) -> MLReader:
+        """Uunsupported."""
         raise NotImplementedError(
             f"{cls} does not support loading/loading, just re-fit the estimator to re-create a model."
```
Comment (lijinf2) [inline python/src/spark_rapids_ml/knn.py:469]: Sorry can you help fix my typo "loading/loading" to "saving/loading".

File: python/src/spark_rapids_ml/knn.py [hunk 4]
```diff
@@ -442,16 +457,31 @@ def exactNearestNeighborsJoin(
...
         )
 
+    def save(self, path: str) -> None:
+        """Unsupported."""
+        raise NotImplementedError(
+            "ApproximateNearestNeighbors does not support saving/loading, just re-create the estimator."
```
Comment (lijinf2) [inline python/src/spark_rapids_ml/knn.py:475]: Should we replace "ApproximateNearestNeighbors" with `f"{self.__class__}"` as the function is shared between NN and ANN?

File: python/src/spark_rapids_ml/knn.py [hunk 5]
```diff
@@ -442,16 +457,31 @@ def exactNearestNeighborsJoin(
...
+        )
+
+    @classmethod
+    def load(cls, path: str) -> MLReader:
+        """Unsupported."""
+        raise NotImplementedError(
+            "ApproximateNearestNeighbors does not support saving/loading, just re-create the estimator."
```
Comment (lijinf2) [inline python/src/spark_rapids_ml/knn.py:482]: Same here, should we replace "ApproximateNearestNeighbors" with f"{cls}"?

**Review by lijinf2 — COMMENTED** [review]
Overall looks good. Minor comments.

**Comment by eordentlich** [comment]
build

File: python/src/spark_rapids_ml/knn.py (hunk 5 above)
Comment (eordentlich) [reply python/src/spark_rapids_ml/knn.py]: thanks for catching all above.

**Review by lijinf2 — APPROVED** [review]
Looks good!

**Pull request merged**